vta/apps/tsim_example/CMakeLists.txt

```diff
@@ -35,8 +35,12 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND
 endif()
 
 file(GLOB TSIM_SW_SRC src/driver.cc)
+list(APPEND TSIM_SW_SRC ${VTA_DIR}/src/vmem/virtual_memory.cc)
+list(APPEND TSIM_SW_SRC ${VTA_DIR}/src/dpi/module.cc)
+
 add_library(sw SHARED ${TSIM_SW_SRC})
-target_include_directories(sw PRIVATE ${VTA_DIR}/include)
+target_compile_definitions(sw PUBLIC USE_TSIM)
```

Comment:
Just out of curiosity, I've seen two places using this macro, one is to define `vta_phy_addr_t`, the other is to define `VTADeviceRun`. What is the reason for adding this in `tsim_example`?

Reply:
You are right, now that we have the virtual memory manager we don't need this for simulation. I am going to remove it from the example.

`USE_TSIM` is basically a quickfix for FPGAs with 64-bit addresses, although I don't see any reason why not using virtual memory manager on actual devices and getting rid of this.

What is your take on using vmem on actual devices and not just in simulation?

Reply:
I think using vmem on actual hardware might drop the performance. Instead, implementing a `MMU` to replace the software vmem would be a better choice.